tests/lib/bin/invariant-tool

```diff
@@ -0,0 +1,78 @@
+#!/bin/sh
+if [ $# -eq 0 ]; then
+	echo "usage: invariant-tool check [INVARIANT...]"
+	exit 1
+fi
+
+action=
+while [ $# -gt 0 ]; do
+	case "$1" in
+		--)
+			shift
+			break
+			;;
+		check)
+			action=check
+			shift
+			;;
+		-*)
+			echo "invariant-tool: unsupported argument $1" >&2
+			exit 1
+			;;
+		*)
+			break
+			;;
+	esac
+done
+
+check_root_files_in_home() {
+	n="$1" # invariant name
+	find /home -mindepth 1 -user root -a ! -path "$GOHOME" -a ! -path "$GOHOME/*" 2>/dev/null >"/tmp/invariant-tool.$n"
```

Comment:
I wonder if this could be simplified by just doing find over the only test user we have in the test (or do we have more?) and gettting rid of `-path ...` exclusions.

Reply:
Yes but:
 - we have a special test that adds a test-zsh user, that would not catch it
 - we have the magic /home/ubuntu that picked up a what looks like, a real bug with cloud-init

Reply:
Yes, you're right, I didn't know of extra users.